src/main/java/com/phamhieu/bookapi/api/user/UserController.java

```diff
@@ -0,0 +1,57 @@
+package com.phamhieu.bookapi.api.user;
+
+import com.phamhieu.bookapi.domain.user.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.UUID;
+
+import static com.phamhieu.bookapi.api.user.UserDTOMapper.*;
+
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/users")
+public class UserController {
+
+    private final UserService userService;
+
+    @Operation(summary = "Find all users")
+    @GetMapping
+    public List<UserResponseDTO> findAll() {
+        return toUserDTOs(userService.findAll());
+    }
+
+    @Operation(summary = "Find user by id")
+    @GetMapping("{userId}")
+    public UserResponseDTO findUserById(@PathVariable UUID userId) {
+        return toUserResponseDTO(userService.findById(userId));
+    }
+
+    @Operation(summary = "Find user by name")
+    @GetMapping("search")
+    public List<UserResponseDTO> find(final @RequestParam String name) {
+        return toUserDTOs(userService.find(name));
+    }
+
+    @Operation(summary = "Add new user")
+    @PostMapping
+    public UserResponseDTO add(@RequestBody UserRequestDTO userRequestDTO) throws NoSuchAlgorithmException {
```

Comment:
Đừng dùng nhiều từ khác nhau cho cùng 1 khái niệm

```suggestion
    public UserResponseDTO create(@RequestBody UserRequestDTO userRequestDTO) throws NoSuchAlgorithmException {
```